daemon/containerd/image_commit.go

```diff
@@ -117,7 +117,17 @@ func (i *ImageService) CommitImage(ctx context.Context, cc backend.CommitConfig)
 			return "", fmt.Errorf("failed to create new image: %w", err)
 		}
 	}
-	return image.ID(img.Target.Digest), nil
+	id := image.ID(img.Target.Digest)
+
+	c8dImg, err := i.NewImageManifest(ctx, img, commitManifestDesc)
+	if err != nil {
+		return id, err
+	}
+	if err := c8dImg.Unpack(ctx, container.Driver); err != nil && !cerrdefs.IsAlreadyExists(err) {
+		return id, fmt.Errorf("failed to unpack image: %w", err)
```

Comment:
not a blocker, but just noticing this; should we return an empty `id` in both of these cases? (as we're now returning an `id` _AND_ and `error`)?

Reply:
> IMO, considering that this function is a bit "lower-level", returning both a non-empty id and a non-nil error is consistent with what happened.

Yeah, it's not a blocker for sure. Mostly trying to avoid some code, somewhere ignoring the error, and depending on just the `id`.

> Potentially this could be a non-hard error depending on the use case. For example when doing the docker commit, unpacking failure could be still "fixable".

Maybe the function has too many responsibilities (i.e., it's not fully atomic, and "part of it" can fail).

Either way; not important, but mostly something I noticed. It may be one to keep in the back of our mind for improving in future.